src/_forms.scss

```diff
@@ -0,0 +1,259 @@
+/**
+ * Outer wrapper class for all input fieldsets
+ */
+.field {
+    margin: 1.5em 0;
+}
+
+/**
+ * Wrapper class for all input fields
+ */
+.input, 
+.textarea, 
+.select {
+    margin: 0;
+    position: relative;
+    overflow: hidden;
+
+    /**
+    * Underline for input fields
+    */
+    &::after, 
+    &::before {
+        content: '';
+        width: 100%;
+        border-bottom: 2px solid #222;
+        position: absolute;
+        bottom: 0;
+        left: 0;
+    }
+
+    /**
+    * After pseudo element for input underlines
+    * initially hidden outside by parent class
+    * overflow: hidden
+    * 1. Move the pseudo element to the left by 100%
+    * 2. Add animation effects to transform style
+    */
+    &::after {
+        margin-top: 1px;
+        border-bottom: 4px solid #51AFC1;
+        -webkit-transform: translate3d(-101%, 0, 0); /* 1 */
+	            transform: translate3d(-101%, 0, 0); /* 1 */
+        -webkit-transition: -webkit-transform 0.3s;  /* 2 */
+                transition:         transform 0.3s;  /* 2 */
+    }
+
+    /**
+    * After pseudo animated and comes into view
+    * when child input is focussed
+    */
+    &:focus-within::after {
+        -webkit-transform: translate3d(0, 0, 0);
+                transform: translate3d(0, 0, 0);
+    }
+}
+
+/**
+* General styling for the various inputs.
+* The input is placed inside a wrapper with a class
+* of the same name.
+*/
+.input input, 
+.textarea textarea, 
+.select select {
+    height: 2em;
+    width: 100%;
+    border: none;
+    padding: 0;
+    background: transparent;
+    font-size: 1em;
```

Comment:
Do we have to set the font size here? It's probably the same as the body font size